src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/JavaASTAnalyser.java

```diff
@@ -314,6 +319,22 @@ private void tokeniseMavenPom(Pom mavenPom) {
             addToken(INDENT, new Token(PUNCTUATION, "}"), NEWLINE);
         }
 
+        // Maven name
+        String nameDefinitionId = tokeniseKeyValue("name", mavenPom.getName(), "");
+        if (!MAVEN_NAME.matcher(mavenPom.getName()).matches()) {
+            apiListing.addDiagnostic(new Diagnostic(DiagnosticKind.WARNING, "nameDefinitionId",
+                "Maven library name should follow the pattern 'Microsoft Azure client library for <service name>'.",
+                "https://azure.github.io/azure-sdk/java_introduction.html#java-maven-name"));
+        }
+
+        // Maven description
+        String descriptionDefinitionId = tokeniseKeyValue("description", mavenPom.getDescription(), "");
+        if (!MAVEN_DESCRIPTION.matcher(mavenPom.getDescription()).matches()) {
+            apiListing.addDiagnostic(new Diagnostic(DiagnosticKind.WARNING, "descriptionDefinitionId",
+                "Maven library description should follow the pattern 'This package contains the Microsoft Azure <service> client library'.",
+                "https://azure.github.io/azure-sdk/java_introduction.html#java-maven-description"));
+        }
+
```

Comment:
This is not where diagnostics go (they don't go in the JavaASTAnalyser class) - refer to the Diagnostics class.

Reply:
I understand why you want to do it here (because Maven isn't available in diagnostics), but it feels like a conflict to diagnose here...I would rather improve Diagnostics to support this.